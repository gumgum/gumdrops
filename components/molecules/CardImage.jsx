import React from 'react';
import PropTypes from 'prop-types';
import trimString from '../utils/trimString';

const CardImage = ({ url, option, size, className, style }) => {
    const baseClass = 'gds-card__img-container',
        optionClass = option ? `${baseClass}--${option}` : '',
        sizeClass = size ? `${baseClass}--${size}` : '';

    const classNames = trimString(`${baseClass} ${optionClass} ${sizeClass} ${className}`);

    return (
        <div className={classNames} style={style}>
            <img className="gds-card__img" src={url} />
            <div className="gds-card__img-helper" />
        </div>
    );
};

CardImage.displayName = 'CardImage';

CardImage.defaultProps = {
    option: 'top'
};

CardImage.propTypes = {
    url: PropTypes.string.isRequired,
    /** top, bottom */
    option: PropTypes.oneOf(['top', 'bottom']),
    /** xs, sm, md, lg, xl */
    size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
    className: PropTypes.string,
    style: PropTypes.object
};

export default CardImage;
