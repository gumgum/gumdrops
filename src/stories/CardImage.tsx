import React from 'react';
import cx from 'classnames';
import { Sizes } from 'types';

export interface CardImageProps extends React.HTMLAttributes<Element> {
    url: string;
    option?: 'top' | 'bottom';
    size?: Sizes;
}

const baseClass = 'gds-card__img-container';

export const CardImage: React.FC<CardImageProps> = ({
    url,
    option = 'top',
    size,
    className,
    ...otherProps
}) => {
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${option}`]: option,
        [`${baseClass}--${size}`]: size
    });

    return (
        <div className={rootClass} {...otherProps}>
            <img className="gds-card__img" src={url} />
            <div className="gds-card__img-helper" />
        </div>
    );
};
