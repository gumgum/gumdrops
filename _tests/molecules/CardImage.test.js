import renderer from 'react-test-renderer';
import React from 'react';
import CardImage from '../../components/molecules/CardImage';

const defaultProps = {
    option: 'top',
    className: 'custom',
    size: 'xl',
    style: { width: 120 },
    url: 'https://foo.bar'
};

test('Expect <CardImage> to render properly', () => {
    const tree = renderer.create(<CardImage {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
});
