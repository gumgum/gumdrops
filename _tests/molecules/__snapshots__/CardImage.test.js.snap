// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <CardImage> to render properly 1`] = `
<div
  className="gds-card__img-container custom gds-card__img-container--top gds-card__img-container--xl"
  style={
    {
      "width": 120,
    }
  }
>
  <img
    className="gds-card__img"
    src="https://foo.bar"
  />
  <div
    className="gds-card__img-helper"
  />
</div>
`;
